neuralmonkey/encoders/recurrent.py

```diff
@@ -103,6 +106,16 @@ def rnn_layer(rnn_input: tf.Tensor, lengths: tf.Tensor,
         if rnn_spec.cell_type == "LSTM":
             final_state = final_state.h
 
+    # residual commenctions
+    if add_residual:
+        if outputs.get_shape()[-1].value != rnn_input.get_shape()[-1].value:
+            warn("Size of the RNN layer input ({}) and layer output ({}) "
```

Comment:
Tohle se mi nelíbí. U RNN obecně by měly být výstupy stejně velký jako vstupy. 
Jediný, když se dělá bidirectional, ale tam by mi dávalo spíš smysl udělat residual connection tak, že to přičtu ke každý direkci zvlášť a pak to teprve zkonkatenuju.

Možná bych dovolil ty residual jen když neni bidirectional. Jak to dělají ti, podle kterých to implementuješ?

Reply:
No rozhodne v tuto chvili nekontrolujeme, jestli size(vstup) == size(vystup) (protoze rnn bunkam je to jedno). Takze se rozhodnete, jestli tohle chcete zkonstrainovat; pak by se ta kontrola presunula do initu

Reply:
ping - jak to teda vyresolvit?